Sources/SwiftProtobuf/JSONDecodingOptions.swift

```diff
@@ -21,5 +21,9 @@ public struct JSONDecodingOptions {
   /// while parsing.
   public var messageDepthLimit: Int = 100
 
+  /// If unknown fields in the JSON should be ignored. If they aren't
+  /// ignored, an error will be raised if one is encountered.
+  public var ignoreUnknownFields: Bool = false
```

Comment:
Just out of curiosity, why is the default value for this option false?

I ask because it seems more natural to set this to true, to me, as someone who uses protobuf to help manage version compatibility across various components.

Reply:
I also had the pointer in the second commit on the PR, but it is in the spec –

Spec: https://developers.google.com/protocol-buffers/docs/proto3#json_options

> **Ignore unknown fields**: Proto3 JSON parser should reject unknown fields by default but may provide an option to ignore unknown fields in parsing.

With the binary format, the unknown fields can be parsed and preserved to be reflected back out.  But for the JSON format, there is no way to know if something was an int32, int64, uint32, etc., so while it could be reflected back out in JSON, you can't convert it or even expose it via a reflection (if you language has proto reflection).

Reply:
Also, since the "ignore" is destructive, it really shouldn't be the default since it means by default we lose things we can't deal with.  If we tried to capture the text and reflect it out for JSON, maybe it would make sense, but the fact that a binary conversion would loose things is also bad.

There is a similar issue with enums, if you get a value string you don't know, there is no way to keep it.  They also offer up that folks could provide an encoding option to use number, but we haven't added that.

Reply:
Thanks, that makes a lot of sense. I am not accustomed to using JSON with proto aside from debugging.